lib/reporters.js

```diff
@@ -14,6 +14,18 @@ var defaultReporter = function (result) {
   });
   return out.join('\n');
 },
+// Default Partial Grunt reporter
+defaultPartialReporter = function (result, options) {
+  var wrapData = options.wrapData;
```

Comment:
This is still problematic, since a custom reporter would have to check for `options.wrapData` and do the same `lineOffset` calculations. 

How about, since we're already massaging `options.files` anyway, to massage the result, so that `message.lastLine` is adjusted for the `lineOffset`? Then we can use both default and custom reporters without any modifications.